src/pipe.rs

```diff
@@ -0,0 +1,40 @@
+use bytes::{Bytes, BytesMut};
+use futures::sink::{Sink, Wait};
+use futures::sync::mpsc::Sender;
+use std::io::{Error, ErrorKind, Result, Write};
+
+pub struct Pipe {
```

Comment:
Currently, this whole module has no documentation. Could you add some? Even if the purpose seems kind of clear.